qa: header_payload_demux flakiness

From: https://ci.gnuradio.org/buildbot/#/builders/51/builds/331/steps/5/logs/stdio
```
321/360 Test #321: qa_header_payload_demux ............................***Failed    7.95 sec
F......
======================================================================
FAIL: test_001_headerpadding (__main__.qa_header_payload_demux)
Like test 1, but with header padding.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/src/pull/2204/master/gr-digital/python/digital/qa_header_payload_demux.py", line 275, in test_001_headerpadding
    self.assertEqual(payload_sink.data(), payload)
AssertionError: Tuples differ: () != (5, 6, 7, 8, 9, 10, 11, 12, 13...
Second tuple contains 15 additional elements.
First extra element 0:
5
- ()
+ (5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19)
----------------------------------------------------------------------
Ran 7 tests in 3.883s
FAILED (failures=1)
Triggers at: 5 23 31
Random seed: 77172390
Triggers at: 5 23 31
Random seed: 77172390
```

```
339/360 Test #339: qa_header_payload_demux ............................***Failed    6.15 sec
F......
======================================================================
FAIL: test_001_headerpadding (__main__.qa_header_payload_demux)
Like test 1, but with header padding.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/cache/workdir/Ubuntu_16_04_64_remote_0/build_Ubuntu_16_04_64/src/gr-digital/python/digital/qa_header_payload_demux.py", line 274, in test_001_headerpadding
    self.assertEqual(header_sink.data(),  (0,) + header + (payload[0],))
AssertionError: Tuples differ: () != (0, 1, 2, 3, 5)
Second tuple contains 5 additional elements.
First extra element 0:
0
- ()
+ (0, 1, 2, 3, 5)
----------------------------------------------------------------------
Ran 7 tests in 2.783s
FAILED (failures=1)
Triggers at: 5 23 31
Random seed: 77249736
Triggers at: 5 23 31
Random seed: 77249736
```

Build: https://ci.gnuradio.org/buildbot/#/builders/49/builds/405/steps/7/logs/stdio

--- a/gr-digital/python/digital/qa_header_payload_demux.py
+++ b/gr-digital/python/digital/qa_header_payload_demux.py
@@ -103,8 +103,8 @@ class qa_header_payload_demux (gr_unittest.TestCase):
         """Execute self.tb"""
         stop_time = time.time() + timeout
         self.tb.start()
-        while len(payload_sink.data()) < payload_len and \
-                len(header_sink.data()) < header_len and \
+        while (len(payload_sink.data()) < payload_len or \
+                len(header_sink.data()) < header_len) and \
                 time.time() < stop_time:
             time.sleep(.2)
         self.tb.stop()

